internal/topic/core.go

```diff
@@ -58,8 +58,7 @@ func (c *Core) CreateTopic(ctx context.Context, m *Model) error {
 	if err != nil {
 		return err
 	}
-	// TODO: take number of partitions as input
-	_, err = admin.CreateTopic(ctx, messagebroker.CreateTopicRequest{m.Name, 2})
+	_, err = admin.CreateTopic(ctx, messagebroker.CreateTopicRequest{m.Name, m.NumPartitions})
```

Comment:
Need to add validations around `m.NumPartitions` value. 

Just as we have a `DefaultNumPartitions ` value, I think we should define a `MaxAllowedNumPartitions ` value as well to restrict accidental over-provisioning. Thoughts?

Reply:
Yes, we need a lot of validations here, creating a [issue](https://github.com/razorpay/metro/issues/108) for this to fix in separate PR